src/Features/ExternalAccess/OmniSharp/Options/OmniSharpSolutionAnalyzerConfigOptionsUpdater.cs

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.Options;
+using Roslyn.Utilities;
+using Microsoft.CodeAnalysis.ImplementType;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.Options;
+
+using Workspace = CodeAnalysis.Workspace;
+
+internal static class OmniSharpSolutionAnalyzerConfigOptionsUpdater
+{
+    internal static void UpdateOptions(Workspace workspace, OmniSharpEditorConfigOptions editorConfigOptions)
+    {
+        try
+        {
+            workspace.SetCurrentSolution(UpdateOptions, changeKind: WorkspaceChangeKind.SolutionChanged);
+
+            Solution UpdateOptions(Solution oldSolution)
+            {
+                var oldFallbackOptions = oldSolution.FallbackAnalyzerOptions;
+                oldFallbackOptions.TryGetValue(LanguageNames.CSharp, out var csharpFallbackOptions);
+
+                var changedOptions = DetermineChangedOptions(csharpFallbackOptions, editorConfigOptions);
```

Comment:
Do we need to determine changed options? Why not set all the O# options unconditionally?

Reply:
No, that simplifies a lot. Thanks!